graphql_compiler/compiler/subclass.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2017-present Kensho Technologies, LLC.
+from graphql.type.definition import GraphQLObjectType, GraphQLUnionType
+import six
+
+
+def _transitive_closure(graph):
+    """Compute the transitive closure of a graph represented as dict inplace."""
+    # Floyd-Warshall O(N^3)
+    for k, k_out in six.iteritems(graph):
+        for i, i_out in six.iteritems(graph):
+            for j, j_out in six.iteritems(graph):
+                if k in i_out and j in k_out:
+                    graph.setdefault(i, set()).add(j)
+    return graph
+
+
+def compute_subclass_sets(schema, type_equivalence_hints):
```

Comment:
Should this handle the case where `type_equivalence_hints` are not provided? Presumably you can still compute the subclass sets of all non-unions?